compiler/rustc_span/src/def_id/stable_crate_id_map.rs

```diff
@@ -0,0 +1,36 @@
+use super::{CrateNum, StableCrateId};
+
+#[derive(Clone, Default)]
+pub struct Config;
+
+impl odht::Config for Config {
+    type Key = StableCrateId;
+    type Value = CrateNum;
+
+    type EncodedKey = [u8; 8];
+    type EncodedValue = [u8; 4];
+
+    type H = odht::UnHashFn;
+
+    #[inline]
+    fn encode_key(k: &StableCrateId) -> [u8; 8] {
+        k.to_u64().to_le_bytes()
+    }
+
+    #[inline]
+    fn encode_value(v: &CrateNum) -> [u8; 4] {
+        v.as_u32().to_le_bytes()
+    }
+
+    #[inline]
+    fn decode_key(k: &[u8; 8]) -> StableCrateId {
+        StableCrateId(u64::from_le_bytes(*k))
+    }
+
+    #[inline]
+    fn decode_value(v: &[u8; 4]) -> CrateNum {
+        CrateNum::from_u32(u32::from_le_bytes(*v))
+    }
+}
+
+pub type StableCrateIdMap = odht::HashTableOwned<Config>;
```

Comment:
Why `odht`?
Could we use an `IndexSet<StableCrateId, BuildHasherDefault<Unhasher>>`? (`Unhasher` from `rustc_data_structures::unhash`)

Reply:
This is copy pasted from https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/def_path_hash_map/type.DefPathHashMap.html

It seemed similar enough, but I don't think the choice is very important.

Reply:
This worked out fine and is simpler. I squashed it into the existing commit